### DISCUSSION TITLE
Should old congestion.rpt files be left lying around?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#2780] [General]
Old congestion.rpt files are not deleted upon success, which can be confusing.

Is this intentional?

https://github.com/The-OpenROAD-Project/OpenROAD/blob/9d17aaf662bcc3266ff452b7be2e6936f9221e76/src/grt/src/GlobalRouter.cpp#L282

**Comment by maliberty** [comment]
I think this is a bug.

**Reply by oharboe** [reply]
So what is the desired behavior?

It is useful to see congestion.rpt even in a success case to see where the design is hard to route

Perhaps a congestion.rpt that have more categories?

horizontal, vertical and a category to show the hardest parts. This would compliment the heatmap with more specific information.

**Reply by maliberty** [reply]
It is only a report of violations.  The heatmap already has options to show h & v as well as thresholds.

**Reply by oharboe** [reply]
Sounds good. I have not seen that option, but I will look.

**Reply by maliberty** [reply]
double click on the underlined text in the display control

**Reply by oharboe** [reply]
> I think this is a bug.

@maliberty How about: https://github.com/The-OpenROAD-Project/OpenROAD/pull/2786

**Reply by maliberty** [reply]
https://github.com/The-OpenROAD-Project/OpenROAD/issues/2782
